### DISCUSSION TITLE
Compile error (Visual Studio)

### DISCUSSION THREAD

**Discussion by RonnieFilms** [KadeArchive/Kade-Engine#579] [Q&A] [answered]
I am getting an error when compiling the build. The error has to do with windows visual studio, which I have installed. Any help would be appreciated

C:/HaxeToolkit/haxe/lib/polymod/git/polymod/format/XMLMerge.hx:189: characters 10-23 : Warning : This typedef is deprecated in favor of haxe.xml.Access
C:/HaxeToolkit/haxe/lib/polymod/git/polymod/format/XMLMerge.hx:189: characters 30-43 : Warning : This typedef is deprecated in favor of haxe.xml.Access
source/PlayState.hx:299: characters 3-39 : Warning : `FlxCamera.defaultCameras` is deprecated, use `FlxG.cameras.setDefaultDrawTarget` instead

Compiling group: haxe
cl.exe -Iinclude -IC:/HaxeToolkit/haxe/lib/discord_rpc/git/linc/ -IC:/HaxeToolkit/haxe/lib/linc_luajit/0,0,2//linc/ -nologo /WX- /fp:precise -DHX_WINDOWS -GR -O2(optim-std) -Zi(debug) -FdC:\Users\Ronni\Desktop\Kade-Engine-master\Kade-Engine-master\export\release\windows\obj\obj/msvc1964-nc/vc.pdb(debug) -Od(debug) -O2(release) -Os(optim-size) -FS -Oy- -c -EHs -GS- -IC:/HaxeToolkit/haxe/lib/hxcpp/4,2,1/include -DHXCPP_M64 -DHXCPP_VISIT_ALLOCS(haxe) -DHX_SMART_STRINGS(haxe) -DHXCPP_API_LEVEL=400(haxe) -D_CRT_SECURE_NO_DEPRECATE -D_ALLOW_MSC_VER_MISMATCH -D_ALLOW_ITERATOR_DEBUG_LEVEL_MISMATCH -DHX_WIN_MAIN(main) -wd4996 ... tags=[haxe,static]
 - src/io/newgrounds/objects/events/_Result/GetDateTimeResult_Impl_.cpp
 - src/sys/io/_Process/Stdout.cpp
 - src/flixel/util/FlxPool_flixel_effects_FlxFlicker.cpp
 - src/flixel/input/actions/_FlxSteamController/FlxSteamControllerMetadata.cpp
 - src/lime/media/openal/ALC.cpp  [haxe,release]
 - src/lime/system/_ThreadPool/ThreadPoolMessage.cpp
 - src/openfl/display/DOMRenderer.cpp  [haxe,release]
 - src/flixel/system/debug/interaction/tools/Pointer.cpp
 - src/lime/_internal/backend/native/TextEventInfo.cpp
 - src/sys/io/FileOutput.cpp
 - src/flixel/tweens/motion/LinearPath.cpp
 - src/openfl/display/_GradientType/GradientType_Impl_.cpp
 - src/flixel/input/actions/_FlxSteamController/FlxSteamUpdater.cpp
 - src/openfl/system/SecurityDomain.cpp  [haxe,release]
 - src/openfl/display/_internal/Context3DElementType.cpp
Error: GradientType_Impl_.cpp
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29333\include\yvals.h(12): fatal error C1083: Cannot open include file: 'crtdbg.h': No such file or directory

Error: SecurityDomain.cpp
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29333\include\yvals.h(12): fatal error C1083: Cannot open include file: 'crtdbg.h': No such file or directory

Error: DOMRenderer.cpp
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29333\include\yvals.h(12): fatal error C1083: Cannot open include file: 'crtdbg.h': No such file or directory

Error: TextEventInfo.cpp
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29333\include\yvals.h(12): fatal error C1083: Cannot open include file: 'crtdbg.h': No such file or directory

**Comment by Subsonic-Vaporeon** [comment]
The game is built in HAXE, so try there. Are you using Visual Studio or
Visual Studio Code?

On Tue, May 25, 2021 at 5:17 AM RonnieFilms ***@***.***>
wrote:

> I am getting an error when compiling the build. The error has to do with
> windows visual studio, which I have installed. Any help would be appreciated
>
> C:/HaxeToolkit/haxe/lib/polymod/git/polymod/format/XMLMerge.hx:189:
> characters 10-23 : Warning : This typedef is deprecated in favor of
> haxe.xml.Access
> C:/HaxeToolkit/haxe/lib/polymod/git/polymod/format/XMLMerge.hx:189:
> characters 30-43 : Warning : This typedef is deprecated in favor of
> haxe.xml.Access
> source/PlayState.hx:299: characters 3-39 : Warning :
> FlxCamera.defaultCameras is deprecated, use
> FlxG.cameras.setDefaultDrawTarget instead
>
> Compiling group: haxe
> cl.exe -Iinclude -IC:/HaxeToolkit/haxe/lib/discord_rpc/git/linc/
> -IC:/HaxeToolkit/haxe/lib/linc_luajit/0,0,2//linc/ -nologo /WX- /fp:precise
> -DHX_WINDOWS -GR -O2(optim-std) -Zi(debug)
> -FdC:\Users\Ronni\Desktop\Kade-Engine-master\Kade-Engine-master\export\release\windows\obj\obj/msvc1964-nc/vc.pdb(debug)
> -Od(debug) -O2(release) -Os(optim-size) -FS -Oy- -c -EHs -GS-
> -IC:/HaxeToolkit/haxe/lib/hxcpp/4,2,1/include -DHXCPP_M64
> -DHXCPP_VISIT_ALLOCS(haxe) -DHX_SMART_STRINGS(haxe)
> -DHXCPP_API_LEVEL=400(haxe) -D_CRT_SECURE_NO_DEPRECATE
> -D_ALLOW_MSC_VER_MISMATCH -D_ALLOW_ITERATOR_DEBUG_LEVEL_MISMATCH
> -DHX_WIN_MAIN(main) -wd4996 ... tags=[haxe,static]
>
>    - src/io/newgrounds/objects/events/*Result/GetDateTimeResult_Impl*.cpp
>    - src/sys/io/_Process/Stdout.cpp
>    - src/flixel/util/FlxPool_flixel_effects_FlxFlicker.cpp
>    -
>    src/flixel/input/actions/_FlxSteamController/FlxSteamControllerMetadata.cpp
>    - src/lime/media/openal/ALC.cpp [haxe,release]
>    - src/lime/system/_ThreadPool/ThreadPoolMessage.cpp
>    - src/openfl/display/DOMRenderer.cpp [haxe,release]
>    - src/flixel/system/debug/interaction/tools/Pointer.cpp
>    - src/lime/_internal/backend/native/TextEventInfo.cpp
>    - src/sys/io/FileOutput.cpp
>    - src/flixel/tweens/motion/LinearPath.cpp
>    - src/openfl/display/*GradientType/GradientType_Impl*.cpp
>    - src/flixel/input/actions/_FlxSteamController/FlxSteamUpdater.cpp
>    - src/openfl/system/SecurityDomain.cpp [haxe,release]
>    - src/openfl/display/
> *internal/Context3DElementType.cpp Error: GradientType_Impl*.cpp
>    C:\Program Files (x86)\Microsoft Visual
>    Studio\2019\Community\VC\Tools\MSVC\14.28.29333\include\yvals.h(12): fatal
>    error C1083: Cannot open include file: 'crtdbg.h': No such file or directory
>
> Error: SecurityDomain.cpp
> C:\Program Files (x86)\Microsoft Visual
> Studio\2019\Community\VC\Tools\MSVC\14.28.29333\include\yvals.h(12): fatal
> error C1083: Cannot open include file: 'crtdbg.h': No such file or directory
>
> Error: DOMRenderer.cpp
> C:\Program Files (x86)\Microsoft Visual
> Studio\2019\Community\VC\Tools\MSVC\14.28.29333\include\yvals.h(12): fatal
> error C1083: Cannot open include file: 'crtdbg.h': No such file or directory
>
> Error: TextEventInfo.cpp
> C:\Program Files (x86)\Microsoft Visual
> Studio\2019\Community\VC\Tools\MSVC\14.28.29333\include\yvals.h(12): fatal
> error C1083: Cannot open include file: 'crtdbg.h': No such file or directory
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/KadeDev/Kade-Engine/discussions/579>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/APYTVX5UYIW6H6EURZMALBTTPNTKFANCNFSM45O47WPA>
> .
>

-- 
NOTE: The information contained in this email is for the exclusive use of 
the intended recipient(s) and may be confidential, proprietary, and/or 
legally privileged. Inadvertent disclosure of this information does not 
constitute a waiver of any privilege. If you receive this message in error, 
do not directly or indirectly use, print, copy, forward, or disclose any 
part of this message including any attachments. Please delete this email 
and all copies and notify the sender immediately. Return any printed copies 
in person or via US Postal Service to the Office of the Assistant 
Superintendent, SUFSD, 15 Croft Road, Poughkeepsie, NY 12603.

**Reply by RonnieFilms** [reply]
I have both installed, but the error has to do with Visual Studio 2019.

**Comment by puyoxyz** [comment, marked as answer]
Did you install *every part of VS that it requires*? The [building guide has a specific list of what components you need to install in the VS installer](https://kadedev.github.io/Kade-Engine/building)

**Reply by RonnieFilms** [reply]
That's what was causing the problems, thanks.

**Comment by Subsonic-Vaporeon** [comment]
Puyo, I was only asking because she was using Visual Studio and that was
the issue, the code is written in HAXE and only a tiny bit of Lua, so
Ronnie try using those also Visual Studio works with Unity and certain
platforms. HAXE is a little hard to get in to but Lua isn't that advanced.
Besides its only used 0.01% of the time in the code.

On Tue, May 25, 2021 at 7:19 AM Puyo ***@***.***> wrote:

> Did you install *every part of VS that it requires*? The building guide
> has a specific list of what components you need to install in the VS
> installer <https://kadedev.github.io/Kade-Engine/building>
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/KadeDev/Kade-Engine/discussions/579#discussioncomment-780640>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/APYTVX4VUPTKEP64MLCFIATTPOBVHANCNFSM45O47WPA>
> .
>

-- 
NOTE: The information contained in this email is for the exclusive use of 
the intended recipient(s) and may be confidential, proprietary, and/or 
legally privileged. Inadvertent disclosure of this information does not 
constitute a waiver of any privilege. If you receive this message in error, 
do not directly or indirectly use, print, copy, forward, or disclose any 
part of this message including any attachments. Please delete this email 
and all copies and notify the sender immediately. Return any printed copies 
in person or via US Postal Service to the Office of the Assistant 
Superintendent, SUFSD, 15 Croft Road, Poughkeepsie, NY 12603.

**Reply by puyoxyz** [reply]
That has nothing to do with it -- Visual Studio needs to be installed for its compilers, the actual IDE is not the reason it is required. You can edit the code and build it using any IDE or terminal as long as Visual Studio is installed.
